woocommerce/woocommerce
# Blocks E2E: Refactor blocks global setup

### Changes proposed in this Pull Request:

Use the available utilities to streamline the blocks' E2E global setup:
- Replace all browser-based logins from the global setup with REST calls.
- Skip creating a guest storage state file as it's not needed.
- Add some info logs with timers for better DevX.
- Save storage state files along with the test results to include them in the artifacts zip.
- Make the change backward-compatible with the current tests.

The main advantage of this update is the improved stability and less waiting until the actual tests start running. Here are some rough numbers from my machine:

**Initial run** 

| Metric                                      | Old Setup | New Setup | Improvement (%) |
|---------------------------------------------|-----------|-----------|-----------------|
| Authentication Time (s)                     | 9.3       | 4.7       | ~50%             |
| Attributes Preparation Time (s)             | 6.8       | 6.2       | ~9%              |
| Total Time (s)                              | 16.1      | 10.9      | ~32%             |

**Subsequent runs**

| Metric                                      | Old Setup | New Setup | Improvement (%) |
|---------------------------------------------|-----------|-----------|-----------------|
| Authentication Time (s)                     | 5.0       | 1.1       | ~78%            |
| Attributes Preparation Time (s)             | 6.6       | 5.2       | ~21%            |
| Total Time (s)                              | 11.6      | 6.3       | ~45%            |

It might just be seconds, but when restarting tests time and time again, it makes a big difference. 😄 


### How to test the changes in this Pull Request:

Run any `woocommerce-blocks` E2E test a couple of times and ensure the global setup works as expected. All the CI tests should pass.
